### PR TITLE
Add and make local network as default chainid

### PR DESCRIPTION
**Pull request by baichuan3**
## Summary

1. Add local network, the chain id is `20230104`
2. Make the local network as default chain id, and use temporary directory for store
3. Make local chain id as default env
4. Change all dev to test chain id for integration test and other tests
5. Fix testnet url link by the way @feliciss, relative your pr https://github.com/rooch-network/rooch/pull/852

**Before to use new local network, suggest  clean store dir and keystore file.**

default use local network
```
rooch server start
```

use dev network
```
rooch server start -n dev
```

use local dev chainid env
```
rooch env add --chain-id 20230103 --alias devnet --rpc http://0.0.0.0:50051/
rooch env switch --alias devnet
```

### CODE REVIEW

File: crates/rooch-types/src/chain_id.rs [hunk 1]
```diff
@@ -178,6 +198,11 @@ impl BuiltinChainID {
     pub fn genesis_ctx(&self) -> GenesisContext {
         let chain_id = self.chain_id().id;
         match self {
+            BuiltinChainID::Local => {
+                //Local timestamp from 0, developer can manually set the timestamp
```
Comment (feliciss) [inline crates/rooch-types/src/chain_id.rs:202]: Shall it be fetched in local timestamp in UTC?

File: crates/rooch-types/src/chain_id.rs [hunk 2]
```diff
@@ -178,6 +198,11 @@ impl BuiltinChainID {
...
+                let timestamp = 0;
+                GenesisContext::new(chain_id, timestamp)
+            }
             BuiltinChainID::Dev => {
                 //Dev timestamp from 0, developer can manually set the timestamp
```
Comment (feliciss) [inline crates/rooch-types/src/chain_id.rs:207]: This should be dev chain's timestamp.

File: crates/rooch-types/src/chain_id.rs [hunk 3]
```diff
@@ -365,7 +392,11 @@ impl RoochChainID {
     }
 
     pub fn is_test_or_dev(&self) -> bool {
```
Comment (feliciss) [inline crates/rooch-types/src/chain_id.rs:394]: The name needs to be changed.

**Review by feliciss — CHANGES_REQUESTED** [review]
1. Command `rooch server start -dev` will be init a store in:

> StoreConfig init store dir "ev/local/roochdb/rooch_store" "ev/local/roochdb/moveos_store"

This is a bug.

2. Command `rooch server start -n dev` and `rooch server start -n test` shouldn't start with local RPC, as it would violate the rules MetaMask defines if user already added local RPC network on MetaMask:

> This URL is currently used by the Rooch Local Network network.

3. When adding a new environment, verify the `rpc`, `alias` and `chain_id` don't collapse with existing ones. 
4. Once a new env is added, the chain id is still shown as the last element and it's not updated. For example:

> Chain ID
> 20230103
> The RPC URL you have entered returned a different chain ID (20230102). Please update the Chain ID to match the RPC URL of the network you are trying to add.

Config:

```zsh
envs:
- chain_id: 20230104
  alias: default
  rpc: http://0.0.0.0:50051
  ws: null
- chain_id: 20230103
  alias: dev
  rpc: https://devnet-seed.rooch.network/
  ws: null
- chain_id: 20230102
  alias: test
  rpc: https://testnet-seed.rooch.network/
  ws: null
- chain_id: 20230103
  alias: devnet
  rpc: http://0.0.0.0:50051/
  ws: null
active_env: devnet
```

5. After switching to `devnet`, command `rooch server start -n test` and `rooch server start -n dev` result in a panic error:

> thread 'main' panicked at 'Invalid sequencer order', crates/rooch- sequencer/src/actor/sequencer.rs:45:24

6. After a new env is added, starting rooch server will not be added a custom chain id:

> rooch server start -n devnet
> error: Invalid value "devnet" for '--chain-id <CHAIN_ID>': Invalid Custom chain id devnet, custom chain id format is: chain_name:chain_id

**Comment by baichuan3** [comment]
I have also encountered problems related to env. There are some problems with the current experience. There have been discussions in diccord. I started a separate PR (https://github.com/rooch-network/rooch/issues/857) to handle it. The current PR only handles the local network and related modifications.

**Comment by baichuan3** [comment]
> 1. Command `rooch server start -dev` will be init a store in:
> 
> > StoreConfig init store dir "ev/local/roochdb/rooch_store" "ev/local/roochdb/moveos_store"
> 
> This is a bug.
> 
> 2. Command `rooch server start -n dev` and `rooch server start -n test` shouldn't start with local RPC, as it would violate the rules MetaMask defines if user already added local RPC network on MetaMask:
> 
> > This URL is currently used by the Rooch Local Network network.
> 
> 3. When adding a new environment, verify the `rpc`, `alias` and `chain_id` don't collapse with existing ones.
> 4. Once a new env is added, the chain id is still shown as the last element and it's not updated. For example:
> 
> > Chain ID
> > 20230103
> > The RPC URL you have entered returned a different chain ID (20230102). Please update the Chain ID to match the RPC URL of the network you are trying to add.
> 
> Config:
> 
> ```shell
> envs:
> - chain_id: 20230104
>   alias: default
>   rpc: http://0.0.0.0:50051
>   ws: null
> - chain_id: 20230103
>   alias: dev
>   rpc: https://devnet-seed.rooch.network/
>   ws: null
> - chain_id: 20230102
>   alias: test
>   rpc: https://testnet-seed.rooch.network/
>   ws: null
> - chain_id: 20230103
>   alias: devnet
>   rpc: http://0.0.0.0:50051/
>   ws: null
> active_env: devnet
> ```
> 
> 5. After switching to `devnet`, command `rooch server start -n test` and `rooch server start -n dev` result in a panic error:
> 
> > thread 'main' panicked at 'Invalid sequencer order', crates/rooch- sequencer/src/actor/sequencer.rs:45:24
> 
> 6. After a new env is added, starting rooch server will not be added a custom chain id:
> 
> > rooch server start -n devnet
> > error: Invalid value "devnet" for '--chain-id <CHAIN_ID>': Invalid Custom chain id devnet, custom chain id format is: chain_name:chain_id

1. `rooch server start -dev` is not a bug. When the -d parameter is specified when starting the server, the data-dri specified by the -d parameter will be used as the storage path. When the -d parameter is `-dev`, it indicates the `ev/` path in the current directory. If you want to start the dev network, use the command line `rooch server start -n dev`
2. Other issues related to env will be discussed and resolved in a separate pr https://github.com/rooch-network/rooch/issues/857.

**Comment by feliciss** [comment]
> 1. `rooch server start -dev` is not a bug. When the -d parameter is specified when starting the server, the data-dri specified by the -d parameter will be used as the storage path. When the -d parameter is `-dev`, it indicates the `ev/`

Regarding this, the `-d` flag should leave a whitespace to accept params for the `--data-dir` such as `-d dev` and I think mistyping could result in a wrong store dir. So this is like a bug.

**Comment by baichuan3** [comment]
> > 1. `rooch server start -dev` is not a bug. When the -d parameter is specified when starting the server, the data-dri specified by the -d parameter will be used as the storage path. When the -d parameter is `-dev`, it indicates the `ev/`
> 
> Regarding this, the `-d` flag should leave a whitespace to accept params for the `--data-dir` such as `-d dev` and I think mistyping could result in a different store dir. So this is like a bug.

This is the syntax of `clap`, like `rooch server start -ndev` is equals to `rooch server start -n dev`, Unless we change the implementation of clap, I don't think it's necessary.

**Comment by feliciss** [comment]
> > > 1. `rooch server start -dev` is not a bug. When the -d parameter is specified when starting the server, the data-dri specified by the -d parameter will be used as the storage path. When the -d parameter is `-dev`, it indicates the `ev/`
> > 
> > 
> > Regarding this, the `-d` flag should leave a whitespace to accept params for the `--data-dir` such as `-d dev` and I think mistyping could result in a different store dir. So this is like a bug.
> 
> This is the syntax of `clap`, like `rooch server start -ndev` is equals to `rooch server start -n dev`, Unless we change the implementation of clap, I don't think it's necessary.

Is it possible to define an enum argument?

**Comment by baichuan3** [comment]
> > > > 1. `rooch server start -dev` is not a bug. When the -d parameter is specified when starting the server, the data-dri specified by the -d parameter will be used as the storage path. When the -d parameter is `-dev`, it indicates the `ev/`
> > > 
> > > 
> > > Regarding this, the `-d` flag should leave a whitespace to accept params for the `--data-dir` such as `-d dev` and I think mistyping could result in a different store dir. So this is like a bug.
> > 
> > 
> > This is the syntax of `clap`, like `rooch server start -ndev` is equals to `rooch server start -n dev`, Unless we change the implementation of clap, I don't think it's necessary.
> 
> Is it possible to define an enum argument? That way, the input contents are limited.

Maybe it's an approach.

If we want to strongly verify `-d`, disabling the `short = 'd'` mode and forcing the use of `long = "data-dir"` may be a solution, which will increase the user cost to a certain extent.

File: crates/rooch-config/src/lib.rs [hunk 4]
```diff
@@ -119,14 +119,14 @@ impl BaseConfig {
         let base_data_dir = match opt.base_data_dir.clone() {
             Some(base_data_dir) if base_data_dir.to_str() == Some("TMP") => temp_dir(),
             Some(base_data_dir) => DataDirPath::PathBuf(base_data_dir),
-            // only dev mode use temp store dir
-            None if chain_id.is_dev() => temp_dir(),
+            // only local network use temp store dir
+            None if chain_id.is_local() => temp_dir(),
```
Comment (jolestar) [inline crates/rooch-config/src/lib.rs:123]: Do not use temp_dir as default. In the test case, we can use `TMP` as data dir to use temp_dir.

File: crates/rooch-rpc-client/src/client_config.rs [hunk 5]
```diff
@@ -15,7 +15,8 @@ use serde_with::serde_as;
 use std::fmt::{Display, Formatter, Write};
 
 pub const DEFAULT_EXPIRATION_SECS: u64 = 30;
-pub const ROOCH_TEST_NET_URL: &str = "https://seed-testnet.rooch.network/";
+pub const ROOCH_DEV_NET_URL: &str = "https://devnet-seed.rooch.network/";
```
[the remark this answers is not in the document]
Comment (jolestar) [reply crates/rooch-rpc-client/src/client_config.rs:18]: https://dev-seed.rooch.network

File: crates/rooch-rpc-client/src/client_config.rs [hunk 6]
```diff
@@ -15,7 +15,8 @@ use serde_with::serde_as;
...
+pub const ROOCH_TEST_NET_URL: &str = "https://testnet-seed.rooch.network/";
```
Comment (jolestar) [inline crates/rooch-rpc-client/src/client_config.rs:19]: https://test-seed.rooch.network

File: crates/rooch-types/src/chain_id.rs (hunk 1 above)
Comment (jolestar) [reply crates/rooch-types/src/chain_id.rs:202]: Using local time will cause an undetermined genesis package.

File: crates/rooch-types/src/chain_id.rs (hunk 2 above)
Comment (jolestar) [reply crates/rooch-types/src/chain_id.rs:207]: Set as the test network, use a Ethereum test net block height timestamp.

File: crates/rooch-config/src/lib.rs (hunk 4 above)
Comment (baichuan3) [reply crates/rooch-config/src/lib.rs:123]: default use `local` dir?

Comment (baichuan3) [reply crates/rooch-config/src/lib.rs:123]: If the default network is `local`, it is reasonable to use a temporary directory.

Comment (jolestar) [reply crates/rooch-config/src/lib.rs:123]: `local` is for DApp developer. Most DApp developer wants to keep the data in developing progress. The framework or chain developer wants temp dir. 

DApp developer priority over chain developer.

**Review by jolestar — APPROVED** [review]

File: crates/rooch-config/src/lib.rs (hunk 4 above)
Comment (jolestar) [reply crates/rooch-config/src/lib.rs:123]: We can discuss this tomorrow.

**Comment by jolestar** [comment]
I merged this PR first.

**Pull request merged**